### PR TITLE
Add Pod QOS Guaranteed tests for critical jobs

### PR DESCRIPTION
**Pull request by spiffxp**
Logf instead of Errorf to start with, so we can get a list of jobs that
aren't yet passing

This isn't the most user-friendly way of displaying a report, but this is what I would consider the authoritative list of jobs that have yet to match the new policy

### CODE REVIEW

File: config/tests/jobs/jobs_test.go [hunk 1]
```diff
@@ -979,12 +977,78 @@ func isPodQOSGuaranteed(spec *coreapi.PodSpec) bool {
...
+	if !ok {
+		return false
+	}
+	return re.MatchString(dashboards)
+}
+
+// TODO: s/Should/Must and s/Logf/Errorf when all jobs pass
+func TestKubernetesMergeBlockingJobsShouldHavePodQOSGuaranteed(t *testing.T) {
+	repo := "kubernetes/kubernetes"
+	jobs := c.AllStaticPresubmits([]string{repo})
```
Comment (MushuEE) [inline config/tests/jobs/jobs_test.go:1023]: Is this not also captured in `allStaticJobs?`

File: config/tests/jobs/jobs_test.go [hunk 2]
```diff
@@ -979,12 +977,78 @@ func isPodQOSGuaranteed(spec *coreapi.PodSpec) bool {
...
+		for _, err := range errs {
+			t.Logf("%v (%v): %v", job.Name, branches, err)
+		}
+	}
+}
+
+// TODO: s/Should/Must and s/Logf/Errorf when all jobs pass
+func TestKubernetesReleaseBlockingJobsShouldHavePodQOSGuaranteed(t *testing.T) {
+	for _, job := range allStaticJobs() {
+		// Only consider Pods that are merge-blocking
```
Comment (MushuEE) [inline config/tests/jobs/jobs_test.go:1043]: `release-blocking`

File: config/tests/jobs/jobs_test.go (hunk 1 above)
Comment (spiffxp) [reply config/tests/jobs/jobs_test.go:1023]: `JobBase` doesn't provide info about what repo that job is for if it's a presubmit or a postubmit, so if I use `allStaticJobs` I have no way of filtering for jobs that run against "kubernetes/kubernetes"

**Pull request merged**